### PR TITLE
Make user search do a bit better on word boundary

### PR DESCRIPTION
**Pull request by dbkr**
Fixes https://github.com/vector-im/riot-web/issues/2928

### CODE REVIEW

File: src/components/views/dialogs/ChatInviteDialog.js [hunk 1]
```diff
@@ -27,6 +27,10 @@ var Modal = require('../../../Modal');
 
 const TRUNCATE_QUERY_LIST = 40;
 
+function escapeRegExp(str) {
+  return str.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&");
```
Comment (ara4n) [inline src/components/views/dialogs/ChatInviteDialog.js]: wow that's horrific. might be worth crediting where it came from (plus the indenting is wrong).

File: src/components/views/dialogs/ChatInviteDialog.js [hunk 2]
```diff
@@ -315,13 +319,18 @@ module.exports = React.createClass({
...
-            if (parts[i].indexOf(query) === 0) {
-                return true;
-            }
+        // Try to find the query following a "word boundary", except that
+        // this does avoids using \b because it only considers letters from
+        // the roman alphabet to be word characters.
+        // Instead, we look for the query following either:
+        //  * The start of the string
+        //  * Whitespace, or
+        //  * A fixed number of punctuation characters
```
Comment (ara4n) [inline src/components/views/dialogs/ChatInviteDialog.js]: this list of punctuation chars feels weirdly anaemic - i'd expect brackets, @, etc?

File: src/components/views/dialogs/ChatInviteDialog.js (hunk 1 above)
Comment (dbkr) [inline src/components/views/dialogs/ChatInviteDialog.js]: Done

File: src/components/views/dialogs/ChatInviteDialog.js (hunk 2 above)
Comment (dbkr) [inline src/components/views/dialogs/ChatInviteDialog.js]: I was keeping it fairly light but I guess for no particular reason - added a load.

**Comment by ara4n** [comment]
lgtm

**Pull request merged**